### PR TITLE
http-server: GetOfferings and GetExchanges

### LINKED ISSUE #24: Implement getExchanges handler

### LINKED ISSUE #25: Implement getOfferings handler

### PR DESCRIPTION
**Pull request by jiyoontbd**
closes #24 
closes #25

### CODE REVIEW

File: httpserver/src/main/kotlin/tbdex/sdk/httpserver/models/ExchangesApi.kt [hunk 1]
```diff
@@ -1,6 +1,12 @@
 package tbdex.sdk.httpserver.models
 
-import tbdex.sdk.protocol.models.*
+import tbdex.sdk.protocol.models.Close
```
Comment (jiyoontbd) [inline httpserver/src/main/kotlin/tbdex/sdk/httpserver/models/ExchangesApi.kt:3]: my import auto format must be different from other folks :(

File: httpserver/src/test/kotlin/tbdex/sdk/httpserver/handlers/GetExchangesTest.kt [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        }
+      }
+    }
+    val response = client.get("/exchanges") {
+      bearerAuth("abc123")
+    }
+
+    assertEquals(HttpStatusCode.OK, response.status)
+
+    // todo: currently failing to map value to GetExchangesResponse since Message is a sealed class with abstract fields
```
Comment (jiyoontbd) [inline httpserver/src/test/kotlin/tbdex/sdk/httpserver/handlers/GetExchangesTest.kt:81]: want to make this test pass, but not sure how to do this given that Message is a sealed class with abstract fields (so it's not possible to instantiate it unless we know the types of Message returned ahead of time)

File: httpserver/src/main/kotlin/tbdex/sdk/httpserver/models/ExchangesApi.kt (hunk 1 above)
Comment (phoebe-lew) [reply httpserver/src/main/kotlin/tbdex/sdk/httpserver/models/ExchangesApi.kt:3]: it is better! I couldn't figure out how to make mine do explicit imports, think it has something to do with my detekt settings being fukt

File: httpserver/src/test/kotlin/tbdex/sdk/httpserver/handlers/GetExchangesTest.kt (hunk 2 above)
Comment (jiyoontbd) [reply httpserver/src/test/kotlin/tbdex/sdk/httpserver/handlers/GetExchangesTest.kt]: fixed with help from @phoebe-lew :D

**Review by phoebe-lew — APPROVED** [review]

File: httpserver/src/test/kotlin/tbdex/sdk/httpserver/handlers/GetExchangesTest.kt [hunk 3]
```diff
@@ -0,0 +1,95 @@
+package tbdex.sdk.httpserver.handlers
+
+import ServerTest
+import TestData
+import io.ktor.client.plugins.auth.*
+import io.ktor.client.plugins.auth.providers.*
+import io.ktor.client.request.*
+import io.ktor.client.statement.*
+import io.ktor.http.*
```
Comment (KendallWeihe) [inline httpserver/src/test/kotlin/tbdex/sdk/httpserver/handlers/GetExchangesTest.kt:9]: FYI more implicit imports on this line and the lines bove, not sure if this is intentional -- I recall one such rule being something like, "_if > N number of imports from the given namespace then wildcard it_"

Commenting on b/c of the commentary [above](https://github.com/TBD54566975/tbdex-kt/pull/123/files#r1433587912)

Comment (phoebe-lew) [reply httpserver/src/test/kotlin/tbdex/sdk/httpserver/handlers/GetExchangesTest.kt:9]: @jiyoontbd I think my IJ messed up your imports when I made changes to the test?

Comment (phoebe-lew) [reply httpserver/src/test/kotlin/tbdex/sdk/httpserver/handlers/GetExchangesTest.kt:9]: @KendallWeihe It's usually a stylistic choice and we just want to be consistent across the codebase. IIRC there are some compilation time gains by using explicit imports vs wildcard, but I haven't personally dug too much into that!

Comment (KendallWeihe) [reply httpserver/src/test/kotlin/tbdex/sdk/httpserver/handlers/GetExchangesTest.kt]: I for one prefer the explicit imports, but yeah beauty is in the eye of the beholder

**Review by KendallWeihe — APPROVED** [review]

**Pull request merged**